packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -244,9 +244,10 @@ exports.createContentTypeNodes = ({
   foreignReferenceMap,
   defaultLocale,
   locales,
+  localeFilter,
 }) => {
   const contentTypeItemId = contentTypeItem.name
-  locales.forEach(locale => {
+  locales.filter(localeFilter || (() => true)).forEach(locale => {
```

Comment:
I think we could filter locales earlier - there are few other places where we iterate through locales (for example when creating asset nodes).

Reply:
Place I'm thinking of is https://github.com/gatsbyjs/gatsby/blob/5a13c3dd3a2be3346218da14ad7d7c924bfc6ba8/packages/gatsby-source-contentful/src/fetch.js#L24. If we filter it there - locale will be filtered before we iterate over them when creating nodes

Reply:
Sounds like a good idea! Will change that!